Add locality params to service registration (#167)
hashicorp/consul-ecs: CHANGELOG.md

   - Add the `meshGateway.healthCheckPort` field which can be hit to determine Envoy's readiness.
* Add the [go-discover](https://github.com/hashicorp/go-discover) binary to the Consul ECS image to better support [cloud auto-join](https://developer.hashicorp.com/consul/docs/install/cloud-auto-join).[[GH-160](https://github.com/hashicorp/consul-ecs/pull/160)]

FEATURES

> * Sets the `AWS_REGION` container environment variable and `AvailabilityZone` attribute of an ECS task meta JSON as locality parameters in Consul service registration's payload. These parameters will only be used to perform locality aware routing only for Enterprise Consul installations. [[GH-167](https://github.com/hashicorp/consul-ecs/pull/167)] 

Reviewer: Suggest changing the wording here to make it more clear that we are using the properties from the ECS task to set the locality params:

```suggestion
* Use the `AWS_REGION` container environment variable and `AvailabilityZone` attribute of an ECS task meta JSON to set the locality parameters in Consul service and proxy registrations. These parameters are used to perform locality aware routing for Consul Enterprise installations. [[GH-167](https://github.com/hashicorp/consul-ecs/pull/167)]
```
